pkg/apis/triggers/v1alpha1/event_listener_types.go

```diff
@@ -55,7 +55,7 @@ type EventListenerSpec struct {
 	ServiceAccountName string                 `json:"serviceAccountName"`
 	Triggers           []EventListenerTrigger `json:"triggers"`
 	ServiceType        corev1.ServiceType     `json:"serviceType,omitempty"`
-	Replicas           int32                  `json:"replicas"`
```

Comment:
should this be a *int32 since it is an optional field?

Reply:
Optional field here will use empty value as 0. In case of pointer, it's nil. So if we want 0 as a valid value for Replicas, we can use *int32, otherwise, int32 is also good enough.

Reply:
even if user provide replicas as `0` it will be overridden to `1` so shall we keep `int32` or shall we move it to `*int32`?

Reply:
After thinking about this, I realize that it would better to have it as `*int32` because it would cover some of the scenarios that we discussed in the Sync meeting. If want EL to not care about number of replicas if it isn't provided. Just up the deployment with a certain number which isn't reflected in the EL.Spec.

Reply:
Modified `int32` to `*int32` this way we can avoid accepting `0` value as part of `spec.replicas`.
Handled same thing in validation